File: Publisher Integration/SPX Mediation/Unified Bidding | iOS Integration/Generic/Native Ad request.m
[SmaatoSDK prebidNativeForAdSpaceId:@"SMAATO_ADSPACE_ID" completion:^(SMAUbBid * _Nullable bid, NSError * _Nullable error) {
         
        if (bid && bid.metaData[@"smaato_ubid"]) {
            NSString *ubId = bid.metaData[@"smaato_ubid"];
             
            SMANativeAd *nativeAd = [[SMANativeAd alloc] init];
            nativeAd.delegate = self;
            SMANativeAdRequest *adRequest = [[SMANativeAdRequest alloc] initWithAdSpaceId:@"SMAATO_ADSPACE_ID"];
             
            NSArray<NSString *> *comps = [ubId componentsSeparatedByString:@":"];
            if ([comps count] > 1) {
                 
                SMAAdRequestParams *requestParam = [[SMAAdRequestParams alloc] init];
                requestParam.ubUniqueId = [comps lastObject];
                [nativeAd loadWithAdRequest:adRequest requestParams:requestParam];
            } else {
                [nativeAd loadWithAdRequest:adRequest];
            }
             
        } else {
            return;
        }
    }];

File: Publisher Integration/SPX Mediation/Unified Bidding | iOS Integration/Generic/Native Ad request.swift
SmaatoSDK.prebidNative(forAdSpaceId: "SMAATO_ADSPACE_ID") {[weak self] ubBid, error in
             
            guard let bid = ubBid ,
                  let bidId = bid.metaData["smaato_ubid"] as? String,
                  let nativeAdRequest = SMANativeAdRequest(adSpaceId:  "SMAATO_ADSPACE_ID") else {
                return
            }
             
            let nativeAd = SMANativeAd()
            nativeAd.delegate = self
             
            let comps = bidId.components(separatedBy: ":")
            if comps.count > 1 {
                let params = SMAAdRequestParams()
                params.ubUniqueId = comps.last
                nativeAd.load(with: nativeAdRequest, requestParams: params)
            } else {
                nativeAd.load(with: nativeAdRequest)
            }
        }